temporian/implementation/numpy/data/event.py

```diff
@@ -89,6 +99,110 @@ def schema(self) -> Event:
             sampling=self.sampling.names,
         )
 
+    @staticmethod
+    def from_dataframe(
+        df: pd.DataFrame,
+        index_names: List[str],
+        timestamp_name: str = "timestamp",
+    ) -> "NumpyEvent":
+        """Function to convert a pandas DataFrame to a NumpyEvent
+
+        Args:
+            df: DataFrame to convert to NumpyEvent
+            index_names: Names of the indexes of the DataFrame
+            timestamp_name: Name for timestamp index. Defaults to "timestamp".
+
+        Returns:
+            NumpyEvent: NumpyEvent created from DataFrame
+        """
+        # check index names and timestamp name are in df columns
+        if (
+            not all(index_name in df.columns for index_name in index_names)
+            or timestamp_name not in df.columns
+        ):
+            raise ValueError(
+                f"Index names {index_names} and timestamp name {timestamp_name}"
+                " are not in DataFrame columns"
+            )
+
+        feature_columns = [
+            column
+            for column in df.columns
+            if column not in index_names + [timestamp_name]
+        ]
+
+        sampling = {}
+        data = {}
+
+        if index_names:
+            group_by_indexes = df.groupby(index_names)
+
+            for group in group_by_indexes.groups:
+                columns = group_by_indexes.get_group(group)
+                timestamp = columns[timestamp_name].to_numpy()
+
+                # Convert group to tuple, useful when its only one value
+                if not isinstance(group, tuple):
+                    group = (group,)
+
+                sampling[group] = timestamp
+                data[group] = [
+                    NumpyFeature(feature, columns[feature].to_numpy())
+                    for feature in feature_columns
+                ]
+        else:
+            timestamp = df[timestamp_name].to_numpy()
+            sampling[()] = timestamp
+            data[()] = [
+                NumpyFeature(feature, df[feature].to_numpy())
+                for feature in feature_columns
+            ]
+
+        numpy_sampling = NumpySampling(index=index_names, data=sampling)
+
+        return NumpyEvent(data=data, sampling=numpy_sampling)
+
+    def to_dataframe(
+        self, timestamp_index_name: str = "timestamp"
+    ) -> pd.DataFrame:
+        """Function to convert a NumpyEvent to a pandas DataFrame
+
+        Args:
+            timestamp_index_name: Name for timestamp index. Defaults to "timestamp".
+
+        Returns:
+            pd.DataFrame: DataFrame created from NumpyEvent
+        """
+        df_index = self.sampling.index + [timestamp_index_name]
+        df_features = self.feature_names
+        columns = df_index + df_features
+
+        df = pd.DataFrame(data=[], columns=columns).set_index(df_index)
```

Comment:
I would make the output dataframe have exactly the same format as the expected format in from_dataframe. Like for from_dataframe, I think having the dataframe be without index might be better.